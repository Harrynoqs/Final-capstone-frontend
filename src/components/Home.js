import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import './styles/home.css';
import { NavLink } from 'react-router-dom';
import * as Remix from 'react-icons/fi';
import Laptopcard from './Laptopcard';

const Home = () => {
  const { laptoplibrary } = useSelector((state) => state.laptops);
  const [currentPage, setCurrentPage] = useState(1);
  const displayPerPage = 2;
  const totalPages = Math.ceil(laptoplibrary.length / displayPerPage);
  const startIndex = (currentPage - 1) * displayPerPage;
  const endIndex = Math.min(startIndex + displayPerPage, laptoplibrary.length);
  const records = laptoplibrary.slice(startIndex, endIndex);

  const prePage = () => {
    if (currentPage > 1) {
      setCurrentPage(currentPage - 1);
    }
  };

  const nextPage = () => {
    if (currentPage < totalPages) {
      setCurrentPage(currentPage + 1);
    }
  };

  useEffect(() => {
    if (currentPage > totalPages) {
      setCurrentPage(totalPages);
    }
  }, [currentPage, totalPages]);

  return (
    <div className="homecontainer">
      <h1>LATEST MODELS</h1>
      <p>Please Select a Laptop Model </p>
      <hr />
      <div className="homepage">
        <NavLink className={currentPage <= 1 ? 'prev disabled' : 'prev'} to="#" onClick={prePage}>
          <Remix.FiTriangle className="triangle" />
        </NavLink>
        {records.map((book) => (
          <Laptopcard key={book.id} lappy={book} />
        ))}
        <NavLink className={currentPage >= totalPages ? 'next disabled' : 'next'} to="#" onClick={nextPage}>
          <Remix.FiTriangle className="next-triangle" />
        </NavLink>
      </div>

    </div>
  );
};
export default Home;
