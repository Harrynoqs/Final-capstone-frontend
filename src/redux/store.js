import { configureStore } from '@reduxjs/toolkit';
import rootReducer from './reducer';

const store = configureStore({
  reducer: rootReducer,
  // Other middleware and store configuration can be added here
});

export default store;
