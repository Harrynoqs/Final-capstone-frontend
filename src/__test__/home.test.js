import React from 'react';
import renderer from 'react-test-renderer'; // Import react-test-renderer
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Home from '../components/Home';
import store from '../redux/store';

test('Home component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </Provider>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
