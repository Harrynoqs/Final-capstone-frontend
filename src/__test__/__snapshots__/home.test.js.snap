// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<section
  className="main-page-section"
>
  <div
    className="heading"
  >
    <h1>
      AWESOME LAPTOPS
    </h1>
  </div>
  <div
    className="divider"
  >
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
    <div
      className="divider-bullet"
    />
  </div>
  <div
    className="no-items-available"
  >
    <div>
       There are no laptops currently available
    </div>
  </div>
</section>
`;
